src/ToricVarieties/ToricLineBundles/attributes.jl

```diff
@@ -60,9 +113,33 @@ export degree
 @doc Markdown.doc"""
     basis_of_global_sections_via_rational_functions(l::ToricLineBundle)
 
-Return a basis of the global sections of the toric line bundle `l`.
+Return a basis of the global sections of the toric line bundle `l` in terms of rational functions.
+
+# Examples
+```jldoctest
+julia> v = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> l = ToricLineBundle(v, [fmpz(2)])
+A toric line bundle on a normal toric variety
+
+julia> basis_of_global_sections_via_rational_functions(l)
+6-element Vector{MPolyQuoElem{fmpq_mpoly}}:
+ x1_^2
+ x2*x1_^2
+ x2^2*x1_^2
+ x1_
+ x2*x1_
+ 1
+```
 """
 @attr function basis_of_global_sections_via_rational_functions(l::ToricLineBundle)
+    if has_attribute(toric_variety(l), :vanishing_sets)
+        tvs = vanishing_sets(toric_variety(l))[1]
+        if contains(tvs, l)
+            return []
```

Comment:
This returns an object of type `Vector{Any}`, which breaks type stability of this function. Though I guess it doesn't matter as `@attr` without a return type is applied to this function.

Reply:
Right. Generally speaking (you raised the exact same point above), this would be a list of polynomials in a multivariate polynomial ring. The coefficients could be `ZZ` or `QQ`. As of now, we even allow any coefficient ring. I am not sure what the corresponding type for this would be in `OSCAR`.

@lkastner Maybe we should restrict the coefficient rings to be either `ZZ` or `QQ`? For all I have in mind, `QQ` should be sufficient. To switch to `ZZ`, one could use the ideal/ring constructors which accept a (coefficient) ring as first argument. In following along these lines, we could even go back an fix the choice of the coefficient ring to be `QQ`.

A natural point to address this is https://github.com/oscar-system/Oscar.jl/issues/1257.

Reply:
I have indeed just added this to https://github.com/oscar-system/Oscar.jl/issues/1257 and hope to fix this soon in a separate PR.

Reply:
I have been thinking about this and probably the coefficient ring should become a template parameter with default `QQ` instead of a property/attribute.

Reply:
Ok. Let us discuss this. (For all it matters, this seems related to https://github.com/oscar-system/Oscar.jl/pull/1274. I hope to propose a fix once this PR is merged. Otherwise, a massive rebase would be necessary.)